### DISCUSSION TITLE
How to boot into a NixOs generated image using iPXE

### DISCUSSION THREAD

**Discussion by leElvyn** [ipxe/ipxe#799] [Q&A] [answered]
Hi, i would like to boot to an iso on an HTTP server via iPXE. I use `nixos-generators` to create images. It supports generating raw EFI files, but they are very large, and i would like my iso to fit within less than 500-700mb. So i generate ISO files, but, as mentioned in previous threads, https://forum.ipxe.org/showthread.php?tid=13148, ISOs can't be booted to since they will try to read sectors on the absent disc. So, how to deliver compressed images that iPXE can run ?
The generated iso seem to contain the required files for EFI boot. 

Here are the features availables on iPXE : 
```
Features: DNS HTTP HTTPS iSCSI TFTP SRP VLAN AoE EFI Menu
```

**Comment by NiKiZe** [comment]
As it says, don't

A proper working ISO for EFI is a vfat (fat32) compatible filesystem, so in theory it might be just fine.
What you do is that you extract the kernel and any needed initrd and boot that directly.

If you also have a squashfs that is need or similar for successful boot then you need to check what is the correct way for your distro.
I prefer and have made sure that Gentoo can append squashfs as a cpio part, and that is used on https://gentoo.ipxe.se (or https://github.com/NiKiZe/Gentoo-iPXE) But other systems require that additional files be downloaded after first part of boot.

You might also want to check any of the existing posts on the topic https://github.com/search?q=repo%3Aipxe%2Fipxe+iso&type=discussions

**Comment by stappersg** [comment]
> would like to boot to an iso on an HTTP server via iPXE.

Haven't done that.  Maybe someday I have a need for it.
For that day I'm adding this comment.

* loop mount the ISO
* https://lists.ipxe.org/pipermail/ipxe-devel/2017-May/005645.html
* https://www.offensive-security.com/kali-linux/booting-kali-linux-live-http/ which has a screenshot of _boot loader command line_
```text
vmlinuz vga=788 initrd=initrd.img append boot=file  components fetch=http://192.168.1.100/filesystem.squashfs
```

And do hope that this discussion get comments with what does work.

**Comment by leElvyn** [comment, marked as answer]
So, for posterity, while i did not yet tried it, this seems like exactly the solution i was looking for after looking into how nixos was netbooted, and how these images are created : https://nixos.wiki/wiki/Netboot. It even produces a chainloadable .ipxe file. 

Thank you for that help !
